src/sqlancer/postgres/ast/PostgresJoin.java

```diff
@@ -18,17 +19,33 @@ public static PostgresJoinType getRandom() {
     private final PostgresTable table;
     private final PostgresExpression onClause;
     private final PostgresJoinType type;
+    private PostgresCTE CTE = null;
```

Comment:
I think it would be cleaner here to have a single field (e.g., `PostgresExpression joinedElement`) which either holds a table or a CTE. This requires implementing an additional class for a table reference, like https://github.com/sqlancer/sqlancer/blob/master/src/sqlancer/tidb/ast/TiDBTableReference.java. Do you think this would be possible?